File: Beecrowd/Beginner/1435 - 1985/1557 - Square Matrix III/C++/1557_SquareMatrixIII.cpp
#include <bits/stdc++.h>

using namespace std;

int main() {
    int arraySize;

    while (cin >> arraySize && arraySize != 0) {
        int matrix[arraySize][arraySize];

        for (int i = 0; i < arraySize; i++) {
            for (int j = 0; j < arraySize; j++) {
                matrix[i][j] = pow(2, i + j);
            }
        }

        int max_width = log10(matrix[arraySize - 1][arraySize - 1]) + 1;

        for (int i = 0; i < arraySize; i++) {
            for (int j = 0; j < arraySize - 1; j++) {
                cout << setw(max_width) << matrix[i][j] << " ";
            }
            
            cout << setw(max_width) << matrix[i][arraySize - 1] << endl;
        }

        cout << endl;
    }

    return 0;
}

File: Beecrowd/Beginner/1435 - 1985/1557 - Square Matrix III/Python/1557_SquareMatrixIII.py
import math

while True:
    array_size = int(input())
    
    if array_size == 0:
        break

    matrix = [[0 for _ in range(array_size)] for _ in range(array_size)]

    for i in range(array_size):
        for j in range(array_size):
            matrix[i][j] = 2 ** (i + j)

    max_width = int(math.log10(matrix[array_size - 1][array_size - 1])) + 1

    for i in range(array_size):
        for j in range(array_size - 1):
            print(f"{matrix[i][j]:>{max_width}}", end=" ")
            
        print(f"{matrix[i][array_size - 1]:>{max_width}}")

    print()
